diff_test.go

```diff
@@ -311,6 +311,18 @@ func TestLog(t *testing.T) {
 	}
 }
 
+func TestSliceType(t *testing.T) {
+	var got string
+	gotp := (*stringPrinter)(&got)
+	a := []struct{}{{}, {}, {}}
+	b := []struct{}{{}, {}}
+	diff.Each(gotp.Printf, a, b)
+	want := "[]struct{}[2:3]: {{}} != {}\n"
+	if got != want {
+		t.Errorf("got = %q, want %q", got, want)
```

Comment:
Since the strings here have indenting and formatting, I like this better (from TestFullField):

```suggestion
		t.Errorf("bad diff")
		t.Logf("got:\n%s", got)
		t.Logf("want:\n%s", want)
```